src/aggregator/aggregator/aggregator.go

```diff
@@ -232,6 +239,46 @@ func (agg *aggregator) AddForwarded(
 	return nil
 }
 
+func (agg *aggregator) AddPassThrough(
+	metric aggregated.Metric,
+	metadata metadata.TimedMetadata,
+) error {
+	callStart := agg.nowFn()
+	agg.metrics.passThrough.Inc(1)
+	agg.RLock()
+	if agg.state != aggregatorOpen {
+		agg.metrics.addPassThrough.ReportError(errAggregatorAlreadyOpenOrClosed)
+		return errAggregatorNotOpenOrClosed
+	}
+	agg.RUnlock()
+
+	if agg.passThroughWriter == nil {
```

Comment:
The way the code is right now, using this var without a (r)lock is racy? e.g. if concurrent `AddPassThrough(...)` and `Close()` calls can interleave to close the writer before this line. 

would be good to add a test to capture this too.